designer/test/section-create.test.js

```diff
@@ -0,0 +1,164 @@
+const { suite, test } = lab
+
+suite('Section create', () => {
+  const data = new Data({
+    sections: [{ name: 'awesomeBadgers' }, { name: 'awesomeBadgers1' }, { name: 'awesomeBadgers2' }]
+  })
+
+  test('Should display form with title and name inputs', () => {
+    const wrapper = shallow(<SectionCreate data={data} />)
+    const form = wrapper.find('form')
+
+    const inputs = form.find('input')
+    expect(inputs.length).to.equal(2)
+    assertTextInput(inputs.at(0), 'section-title', undefined, { value: '' })
+    assertTextInput(inputs.at(1), 'section-name', '')
+  })
+
+  test('Specifying a title should default the name to the camel case version of the string', () => {
+    const wrapper = shallow(<SectionCreate data={data} />)
+
+    wrapper.find('#section-title').simulate('change', { target: { value: 'Bob\'s your uncle' } })
+
+    const inputs = wrapper.find('input')
+    expect(inputs.length).to.equal(2)
+    assertTextInput(inputs.at(0), 'section-title', undefined, { value: 'Bob\'s your uncle' })
+    assertTextInput(inputs.at(1), 'section-name', 'bobsYourUncle')
+  })
+
+  test('Specifying a title should default the name to camel case + an integer when sections with the name already exist', () => {
+    const wrapper = shallow(<SectionCreate data={data} />)
+
+    wrapper.find('#section-title').simulate('change', { target: { value: 'Awesome badgers' } })
+
+    const inputs = wrapper.find('input')
+    expect(inputs.length).to.equal(2)
+    assertTextInput(inputs.at(0), 'section-title', undefined, { value: 'Awesome badgers' })
+    assertTextInput(inputs.at(1), 'section-name', 'awesomeBadgers3')
+  })
+
+  test('Specifying a name should overwrite any auto-generated value', () => {
+    const wrapper = shallow(<SectionCreate data={data} />)
+
+    wrapper.find('#section-title').simulate('change', { target: { value: 'My badgers' } })
+    wrapper.find('#section-name').simulate('blur', { target: { value: 'myName', setCustomValidity: sinon.spy() } })
+    wrapper.find('#section-title').simulate('change', { target: { value: 'Your badgers' } })
+
+    const inputs = wrapper.find('input')
+    expect(inputs.length).to.equal(2)
+    assertTextInput(inputs.at(0), 'section-title', undefined, { value: 'Your badgers' })
+    assertTextInput(inputs.at(1), 'section-name', 'myName')
+  })
+
+  test('Submitting without changing the name should generate the appropriate section', flags => {
+    const clonedData = {
+      addSection: sinon.stub()
+    }
+    data.clone = sinon.stub()
+    data.clone.returns(clonedData)
+
+    const updatedData = sinon.stub()
+    clonedData.addSection.returns(updatedData)
+
+    const savedData = sinon.stub()
+    data.save = sinon.stub()
+    data.save.resolves(savedData)
+
+    const onCreate = data => {
+      expect(data).to.equal(savedData)
+    }
+    const wrappedOnCreate = flags.mustCall(onCreate, 1)
+
+    const wrapper = shallow(<SectionCreate data={data} onCreate={wrappedOnCreate} />)
```

Comment:
same comment from component-create.test.js

Reply:
I *think* this one is probably more sensible - we're expecting a callback to be called after some async processing done on submit. Unfortunately the hapi lab framework doesn't seem to support waiting for a thing to happen and then making some assertions, it only supports waiting for the thing to eventually get called. So this is the only way I can see to assert that the thing is called with the expected args. Here's the documentation for mustCall: https://hapi.dev/module/lab/api/?v=23.0.0#mustcallfunc-count

Reply:
This is now somewhat tidier, thanks @jenbutongit for your suggestion!